providers/os/resources/services/manager.go

```diff
@@ -99,16 +99,21 @@ func ResolveManager(conn shared.Connection) (OSServiceManager, error) {
 			osm = ResolveSystemdServiceManager(conn)
 		}
 	case asset.Platform.Name == "debian":
-		rv := detector.ParseOsVersion(asset.Platform.Version)
-		v, err := rv.MajorAtoi()
-		if err != nil {
-			return nil, errors.New("unknown debian version: " + asset.Platform.Version)
-		}
-
-		if v < 7 {
+		if _, err := conn.FileInfo("/.dockerenv"); err == nil {
+			// The debian docker image does not use systemd
 			osm = &SysVServiceManager{conn: conn}
```

Comment:
Agreed, it is probably more suitable to look for an existing `systemd` package. We may even need to look for a sysv package since that switch to sysv can also happen on VMs https://lecorbeausvault.wordpress.com/2022/02/07/debian-switching-init-system-easily-openrc-sysvinit-runit/

Reply:
do you have any examples of those. Do any of them use systemd?

Reply:
in a vm, we can check `/sbin/init` and see if it points to systemd